partiql-lang/src/main/kotlin/org/partiql/lang/eval/ExprFunction.kt

```diff
@@ -17,10 +17,8 @@ package org.partiql.lang.eval
 import org.partiql.lang.errors.ErrorCode
 import org.partiql.lang.types.FunctionSignature
 
+@Deprecated("As function overloading is now supported, this class is deprecated.", level = DeprecationLevel.ERROR)
 sealed class Arguments
-data class RequiredArgs(val required: List<ExprValue>) : Arguments()
-data class RequiredWithOptional(val required: List<ExprValue>, val opt: ExprValue) : Arguments()
-data class RequiredWithVariadic(val required: List<ExprValue>, val variadic: List<ExprValue>) : Arguments()
```

Comment:
We'll need to retain these -- however, let's deprecate.

Reply:
Since the `sealed class Arguments` has been deprecated, if I retain the 3 lines, we will get errors:
Using 'Arguments' is an error. As function overloading is now supported, this class is deprecated.